linker/linker_test.go

```diff
@@ -1723,7 +1724,7 @@ func TestSyntheticOneOfCollisions(t *testing.T) {
 	}
 	var expected []string
 	require.NotEmpty(t, errs)
-	var actual []string
+	actual := make([]string, 0, len(errs))
 	for _, err := range errs {
 		actual = append(actual, err.Error())
 	}
```

Comment:
We can do even better here
```suggestion
	actual := make([]string, len(errs))
	for i, err := range errs {
		actual[i] = err.Error()
	}
```